packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -0,0 +1,473 @@
+    final RenderAbstractViewport viewport = RenderAbstractViewport.of(contextObject);
+    assert(viewport != null);
+    // If and only if the current scroll offset falls in-between the offsets
+    // necessary to reveal the selected context at the top or bottom of the
+    // screen, then it is already on-screen.
+    final double margin = _dropAreaExtent;
+    final double scrollOffset = _scrollController.offset;
+    final double topOffset = max(
+      _scrollController.position.minScrollExtent, 
+      viewport.getOffsetToReveal(contextObject, 0.0).offset - margin,
+    );
+    final double bottomOffset = min(
+      _scrollController.position.maxScrollExtent,
+      viewport.getOffsetToReveal(contextObject, 1.0).offset + margin,
+    );
+    final bool onScreen = scrollOffset <= topOffset && scrollOffset >= bottomOffset;
+    // If the context is off screen, then we request a scroll to make it visible.
+    if (!onScreen) {
+      _scrolling = true;
+      _scrollController.position.animateTo(
+        scrollOffset < bottomOffset ? bottomOffset : topOffset, 
+        duration: _kScrollAnimationDuration, 
+        curve: Curves.easeInOut,
+      ).then((Null none) {
+        setState(() {
+          _scrolling = false;
+        });
+      });
+    }
+  }
+
+  // Wraps children in Row or Column, so that the children flow in
+  // the widget's scrollDirection.
+  Widget _buildContainerForScrollDirection({List<Widget> children}) {
+    if (widget.scrollDirection == Axis.horizontal) {
+      return new Row(children: children);
+    } 
+    return new Column(children: children);
+  }
+
+  // Wraps one of the widget's children in a DragTarget and Draggable.
+  // Handles up the logic for dragging and reordering items in the list.
+  Widget _wrap(Widget toWrap, int index, BoxConstraints constraints) {
+    assert(toWrap.key != null);
+
+    // Starts dragging toWrap.
+    void onDragStarted() {
+      setState(() {
+        _dragging = toWrap.key;
+        _dragStartIndex = index;
+        _ghostIndex = index;
+        _currentIndex = index;
+        _entranceController.forward(from: 1.0);
+      });
+    }
+
+    // Drops toWrap into the last position it was hovering over.
+    void onDragEnded() {
+      setState(() {
+        if (_dragStartIndex != _currentIndex)
+          widget.onSwap(_dragStartIndex, _currentIndex);
+        _ghostController.reverse(from: 0.1);
+        _entranceController.reverse(from: 0.1);
+        _dragging = null;
+      });
+    }
+
+    Widget buildDragTarget(BuildContext context, List<Key> acceptedCandidates, List<dynamic> rejectedCandidates) {
+      // We build the draggable inside of a layout builder so that we can
+      // constrain the size of the feedback dragging widget.
+      Widget child = new LongPressDraggable<Key>(
+        maxSimultaneousDrags: 1,
+        axis: widget.scrollDirection,
+        data: toWrap.key,
+        feedback: new Container(
+          alignment: Alignment.topLeft,
```

Comment:
If this Container had a key whose value was a GlobalKey created at initState() time, you could get the size with key.currentContext.size in onDragStarted.

Reply:
I tried that and wasn't able to get a non-null Context off the widget. Do we have any examples of this being done somewhere?